middle_end/flambda/from_lambda/cps_conversion.ml

```diff
@@ -246,7 +186,86 @@ let transform_primitive (prim : L.primitive) args loc =
+    | None, None | None, Some _ | Some _, None ->
+      if 1 <= num_dimensions && num_dimensions <= 3 then begin
+        let arity = 1 + num_dimensions in
+        let name = "caml_ba_get_" ^ string_of_int num_dimensions in
+        let desc = Primitive.simple ~name ~arity ~alloc:true in
+        Primitive (L.Pccall desc, args, loc)
+      end else begin
+        Misc.fatal_errorf
+          "Cps_conversion.transform_primitive: \
+           Pbigarrayref with unknown layout and elements should only have \
+           dimensions between 1 and 3 (see translprim)."
+      end
+    end
+  | Pbigarrayset (_unsafe, num_dimensions, kind, layout), args ->
+    begin match C.convert_bigarray_kind kind,
+                C.convert_bigarray_layout layout with
+    | Some _, Some _ ->
+      Primitive (prim, args, loc)
+    | None, None | None, Some _ | Some _, None ->
+      if 1 <= num_dimensions && num_dimensions <= 3 then begin
+        let arity = 2 + num_dimensions in
+        let name = "caml_ba_set_" ^ string_of_int num_dimensions in
+        let desc = Primitive.simple ~name ~arity ~alloc:true in
+        Primitive (L.Pccall desc, args, loc)
+      end else begin
+        Misc.fatal_errorf
+          "Cps_conversion.transform_primimive: \
+           Pbigarrayset with unknown layout and elements should only have \
+           dimensions between 1 and 3 (see translprim)."
+      end
+    end
+  | (Pfloatcomp _ | Psetfield _ | Prevapply | Pdirapply | Pignore | Pidentity
+  | Pbytes_to_string | Pbytes_of_string | Pisint | Pnot | Pnegint | Paddint
+  | Psubint | Pmulint | Pandint | Porint | Pxorint | Plslint | Plsrint | Pasrint
+  | Pcompare_ints | Pcompare_floats | Pintoffloat | Pfloatofint | Pnegfloat
+  | Pabsfloat | Paddfloat | Psubfloat | Pmulfloat | Pdivfloat | Pstringlength
+  | Pstringrefu | Pstringrefs | Pbyteslength | Pbytesrefu | Pbytessetu
+  | Pbytesrefs | Pbytessets | Pflambda_isint | Pgettag | Pisout | Pbswap16
+  | Pint_as_pointer | Popaque | Pgetglobal _ | Psetglobal _ | Pmakeblock _
+  | Pmakefloatblock _ | Pfield _ | Pfield_computed _ | Psetfield_computed _
+  | Pfloatfield _ | Psetfloatfield _ | Pduprecord _ | Pccall _ | Praise _
+  | Pdivint _ | Pmodint _ | Pintcomp _ | Pcompare_bints _ | Poffsetint _
+  | Poffsetref _ | Pmakearray _ | Pduparray _ | Parraylength _ | Parrayrefu _
+  | Parraysetu _ | Parrayrefs _ | Parraysets _ | Pbintofint _ | Pintofbint _
+  | Pcvtbint _ | Pnegbint _ | Paddbint _ | Psubbint _ | Pmulbint _ | Pdivbint _
+  | Pmodbint _ | Pandbint _ | Porbint _ | Pxorbint _ | Plslbint _ | Plsrbint _
+  | Pasrbint _ | Pbintcomp _ | Pbigarraydim _ | Pstring_load_16 _
+  | Pstring_load_32 _ | Pstring_load_64 _ | Pbytes_load_16 _ | Pbytes_load_32 _
+  | Pbytes_load_64 _ | Pbytes_set_16 _ | Pbytes_set_32 _ | Pbytes_set_64 _
+  | Pbigstring_load_16 _ | Pbigstring_load_32 _ | Pbigstring_load_64 _
+  | Pbigstring_set_16 _ | Pbigstring_set_32 _ | Pbigstring_set_64 _ | Pctconst _
+  | Pbbswap _), _ ->
```

Comment:
There are a few primitives in there that we don't want appearing in Ilambda (many of the ones that are transformed, like `Pidentity`, still appear in this pattern).
So in this state, this exhaustive pattern-matching isn't that useful; I'd suggest going back to a `_, _` pattern for this PR and we'll try to have a more useful exhaustive pattern matching in a separate PR.